Automattic/jetpack
# Blocks: Add a new Google Calendar block

This is the first step towards replacing the `[googleapps]` shortcode with blocks.

This PR is currently a work in progress, so please don't be surprised when things don't work. 🙂

Things that aren't working yet:
- Converting from a shortcode block containing the `[googleapps]` shortcode (WordPress/gutenberg#8569 - no progress yet ). 
- Converting a pasted `<iframe>` if user doesn't have UnfilteredHTML permissions (WordPress/gutenberg#18389 - draft PR WordPress/gutenberg#19998, ~~WordPress/gutenberg#18390~~ fixed).

#### Changes proposed in this Pull Request:

Add a new block, `jetpack/google-calendar`.
Add a block version gating arg to jetpack_register_block

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

This is an existing feature being ported to the block editor.

#### Testing instructions:

This is currently marked as a beta block, you will need to add `define( 'JETPACK_BETA_BLOCKS', true );` to your `wp-config.php` file. 

This block also introduces version gating, and you will need wp > 5.3 or gutenberg plugin >= 7.2 to see the block.

* Open the Block Editor.
* Try pasting Google Calendar embed URLs or iframe as an admin user: `https://calendar.google.com/calendar/embed?src=jb4bu80jirp0u11a6niie21pp4%40group.calendar.google.com&ctz=America/New_York` 
or
 `<iframe src="https://calendar.google.com/calendar/embed?src=jb4bu80jirp0u11a6niie21pp4%40group.calendar.google.com&ctz=America/New_York" style="border: 0" width="800" height="600" frameborder="0" scrolling="no"></iframe>` 
* Try editing the embed to a different URL.
* Try converting Classic posts with the Google Calendar `<iframe>` in them.
* Try converting Classic posts with a Google Apps short code in them: `[googleapps domain="calendar" dir="calendar/embed" query="src=jb4bu80jirp0u11a6niie21pp4%40group.calendar.google.com&ctz=America/New_York" width="800" height="600" /]`
* Try accessing the block on WP < 5.4 or gutenberg plugin < 7.2  - it shouldn't be available in versions less than these

![gcal2](https://user-images.githubusercontent.com/3629020/73999145-9f1d3600-49c8-11ea-881f-cdf45d36524f.gif)

#### Proposed changelog entry for your changes:
* Blocks: Add a new Google Calendar block.
